aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1682,6 +1683,60 @@ public void onError(Exception e) {
         };
     }
 
+    /**
+     * Delete the account of the currently logged-in user.
+     * @throws Exception if the user cannot be deleted successfully
+     */
+    @WorkerThread
+    public void deleteUser() throws Exception {
+        final InternalCallback<Void> internalCallback = new InternalCallback<>();
+        internalCallback.await(_deleteUser(internalCallback));
+    }
+
+    /**
+     * Delete the account of the currently logged-in user.
+     * @param callback the callback will be invoked to notify the success or
+     *                 failure of the deleteUser operation
+     */
+    @AnyThread
+    public void deleteUser(final Callback<Void> callback) {
+        final InternalCallback<Void> internalCallback = new InternalCallback<>(callback);
+        internalCallback.async(_deleteUser(internalCallback));
+    }
+    
+    private Runnable _deleteUser(final Callback<Void> callback) {
+        return () -> {
+            if (userpool == null) {
+                callback.onError(new InvalidUserPoolConfigurationException(
+                        "A user pool must be configured in order to delete a user."
+                ));
+            } else {
+                CognitoUser currentUser = userpool.getCurrentUser();
+                currentUser.deleteUserInBackground(new GenericHandler() {
+                    @Override
+                    public void onSuccess() {
+                        signOut(SignOutOptions.builder().signOutGlobally(true).invalidateTokens(true).build(), new Callback<Void>() {
```

Comment:
Just seeing `signOutGlobally` here makes me worry slightly about that doc we saw earlier today about how social sign in `signOutGlobally` doesn't work...  Is that at all applicable here?  Maybe worth a quick check/unit test

Reply:
Although I'm passing the `signOutGlobally(true)` option to `signOut`, a local sign out is performed as well. I tested `deleteUser` with the web UI and it still works as expected.